ddtrace/settings/config.py

```diff
@@ -32,7 +32,11 @@ def __init__(self):
         )
 
         self.env = get_env("env")
+        # get_env will get DD_SERVICE or DATADOG_SERVICE (deprecated)
+        # Note that we do not call self.get_service() here because we never
+        # supported {DD,DATADOG}_SERVICE_NAME for setting the service globally.
```

Comment:
@brettlangdon @jdgumz should we support using `{DD,DATADOG}_SERVICE_NAME` exactly the same as the new `DD_SERVICE`? Currently, I opted to maintain compatibility so there would be no change for existing configurations using `{DD,DATADOG}_SERVICE_NAME`.

This could be confusing either way. Thoughts?

Reply:
Maybe we could add a precedence order? I think this might be baked into `get_env` already, but basically between `DD_` and `DATADOG_` we'd prefer the former over the latter.

That way it doesn't break anything existing, it only "breaks" when you opt into using the new variable format.

Reply:
Yeah the precedence exists in that order already 😄 

The "breaking" I'm thinking of is that we used to only use `{DD,DATADOG}_SERVICE_NAME` for web-framework integrations. If we treat `{DD,DATADOG}_SERVICE_NAME` the same as `{DD,DATADOG}_SERVICE` then existing users will see a whole bunch more collapsed under that service name.

If we keep `.._SERVICE_NAME` and `.._SERVICE` the same then we're only introducing new functionality, not changing anything existing.

Reply:
Ah yeah I see then. We want to keep them separate. `_SERVICE_NAME` should be considered deprecated and will not get the same functionality as `_SERVICE`.